### DISCUSSION TITLE
403 forbidden

### DISCUSSION THREAD

**Discussion by irrisolto** [tweepy/tweepy#1719] [Questions] [answered]
Hi, i have  this code:
```python
import tweepy
data = {
    "bearer": "bearer",
    "api_key": "api key",
    "api_secret": "api secret",
    "access": "access key",
    "access_secret": "access secret"
}
client = tweepy.Client(
    data['bearer'],
    consumer_key=data['api_key'],
    consumer_secret=data['api_secret'],
    access_token=data['access'],
    access_token_secret=data['access_secret'])
user = client.search_all_tweets("from:myuser")
```
and i have this error:
```
Traceback (most recent call last):
  File "C:\Users\User\Desktop\programmazione\Sito felicia\main.py", line 15, in <module>
    user = client.search_all_tweets("from:akaicegirl")
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\tweepy\client.py", line 706, in search_all_tweets
    ), data_type=Tweet
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\tweepy\client.py", line 167, in _make_request
    json=json, user_auth=user_auth)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\tweepy\client.py", line 124, in request
    raise Forbidden(response)
tweepy.errors.Forbidden: 403 Forbidden
```
Why it not works?

**Comment by Harmon758** [comment]
Do you have access to the Academic Research product track?

For code block usage, see https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.

**Comment by irrisolto** [comment]
Is my first issues, idk how it working

**Comment by Harmon758** [comment, marked as answer]
As [the documentation for `Client.search_all_tweets`](https://tweepy.readthedocs.io/en/v4.4.0/client.html#tweepy.Client.search_all_tweets) says, the Twitter API endpoint that it uses, [GET /2/tweets/search/all](https://developer.twitter.com/en/docs/twitter-api/tweets/search/api-reference/get-tweets-search-all) requires you to have been approved for [Academic Research](https://developer.twitter.com/en/products/twitter-api/academic-research) access.
